server/aesdsocket.c

```diff
@@ -27,18 +62,124 @@ void *get_in_addr(struct sockaddr *sa) {
     return &(((struct sockaddr_in6*)sa)->sin6_addr);
 }
 
+/************************************************************************************************
+                        Thread function to handle socket communication
+************************************************************************************************/
+
+void* thread_socket(void* thread_param) {
+
+    char buf[BUF_LEN];
+    char transmit_buffer[BUF_LEN];
+    int fd = open(pathname, O_RDWR | O_CREAT | O_APPEND, S_IWUSR | S_IRUSR | S_IWGRP | S_IRGRP | S_IROTH);
+    thread_data_t *thread_func_param = (thread_data_t *)thread_param;
+    struct sockaddr_storage clientaddr = thread_func_param->clientaddr;
+    char s[INET6_ADDRSTRLEN];
+    inet_ntop(clientaddr.ss_family, get_in_addr((struct sockaddr *)&clientaddr), s, sizeof s);
+    syslog(LOG_USER | LOG_INFO, "Accepted connection from %s\n", s);
+
+    while(1) {
+        memset(buf, '\0', BUF_LEN);
+        size_t recv_bytes = recv(thread_func_param->newsockfd, buf, BUF_LEN, 0);
+        if(recv_bytes == -1) {
+            perror("recv");
+            exit(EXIT_FAILURE);
+        }
+        pthread_mutex_lock(&thread_func_param->mutex);
+        write(fd, buf, recv_bytes);
+        pthread_mutex_unlock(&thread_func_param->mutex);
+        if(buf[recv_bytes-1] == '\n') {
+            break;
+        }
+    }
+    lseek(fd, 0, SEEK_SET);
+    memset(transmit_buffer, '\0', BUF_LEN);
+    int read_bytes = read(fd, transmit_buffer, BUF_LEN);
```

Comment:
@ashwinhr186 what happens if there are more than `BUF_LEN` characters in the file?  Will all packets be sent back to the host?